quickwit-indexing/src/source/kinesis/shard_consumer.rs

```diff
@@ -104,6 +106,7 @@ impl ShardConsumer {
             shutdown_at_shard_eof,
             kinesis_client,
             sink,
+            retry_params: RetryParams::default(),
```

Comment:
Let's add a `retry_params` parameter to the `ShardConsumer` constructor so that the object is inherited from the `KinesisSource`. `RetryParams` is going to need `Clone`.